src/compiler/binder.ts

```diff
@@ -2740,17 +2738,17 @@ namespace ts {
             return expr.parent;
         }
 
-        function lookupSymbolForPropertyAccess(node: EntityNameExpression, lookupContainer: Node = container): Symbol | undefined {
+        function lookupSymbolForPropertyAccess(node: BindableNameExpression, lookupContainer: Node = container): Symbol | undefined {
             if (isIdentifier(node)) {
                 return lookupSymbolForNameWorker(lookupContainer, node.escapedText);
             }
             else {
                 const symbol = lookupSymbolForPropertyAccess(node.expression);
-                return symbol && symbol.exports && symbol.exports.get(node.name.escapedText);
+                return symbol && symbol.exports && symbol.exports.get(getNameOrArgumentText(node));

```

Comment:
wow, this is one of 3 lines with a non-type change in this file. That's crazy!

Reply:
Yup, most of the classification logic actually happens in utilities, and then the same binding processes mostly Just Work™